CRM/Contact/BAO/Query.php

```diff
@@ -6009,10 +6009,13 @@ public function convertToPseudoNames(&$dao, $return = FALSE, $usedForAPI = FALSE
           // and we can use the metadata to figure it out.
           // ideally this bit of IF will absorb & replace all the rest in time as we move to
           // more metadata based choices.
+          $entityName = CRM_Core_DAO_AllCoreTables::getEntityNameForClass($value['bao']);
+          // Get all labels for this pseudoconstant, keyed by id, to avoid a lookup within the loop.
+          $labels[$value['idCol']] ??= array_column(\Civi::entity($entityName)->getOptions($value['idCol']), 'label', 'id');
```

Comment:
I'm guessing that this legacy support in `buildOptions` is what's tripping up the test.

https://github.com/civicrm/civicrm-core/blob/8f164fe29c26c645a33c3e2e489ccc036d379cc5/CRM/Core/DAO.php#L2938-L2942

Tests might be happier with
```suggestion
          $labels[$value['idCol']] ??= $value['bao']::buildOptions($value['idCol']);
```

Reply:
I just tested that and it works.  And really, anything to get out of calling `hook_civicrm_fieldOptions` more than is necessary. I was halfway tempted to load ALL the pseudoconstant option values here outside the top `foreach` loop.

Reply:
> I was halfway tempted to load ALL the pseudoconstant option values here outside the top foreach loop.


Really the `?? =` inside the loop does the same thing.

Reply:
Sorry - I meant load to load all values for all pseudoconstants, at line 5990. But this is probably the better option without knowing more about the other scenarios in which this function is called.